### PR TITLE
Fix  http bug & add binary vectors support

### LINKED ISSUE #1066: it takes too long to insert data to table via rest API
**Describe the bug**
dim=128, num=100000, it cost nearly 40s to insert these vectors to table, need to improve performance here

**Steps/Code to reproduce behavior**
**Expected behavior**
**Environment details**
**Screenshots**
**Additional context**

### PR DESCRIPTION
**Pull request by BossZou**
**What type of PR is this?**
Bug fix

**What this PR does / why we need it:**
Fix bug

**Which issue(s) this PR fixes:**

Fixes #1066, #1067, #1022, #1029, #1028, #1010, #995 

**Special notes for your reviewer:**

**Does this PR introduce a user-facing change?:**

**Additional documentation (e.g. design docs, usage docs, etc.):**

### CODE REVIEW

File: core/src/server/web_impl/controller/WebController.hpp [hunk 1]
```diff
@@ -341,26 +342,26 @@ class WebController : public oatpp::web::server::api::ApiController {
...
-             BODY_DTO(IndexRequestDto::ObjectWrapper, body)) {
+    ENDPOINT("POST", "/tables/{table_name}/indexes", CreateIndex,
+             PATH(String, table_name), BODY_DTO(IndexRequestDto::ObjectWrapper, body)) {
         auto handler = WebRequestHandler();
         handler.RegisterRequestHandler(::milvus::server::RequestHandler());
         auto status_dto = handler.CreateIndex(table_name, body);
+        auto code = status_dto->code->getValue();
 
-        if (0 == status_dto->code->getValue()) {
+        if (0 == code) {
```
Comment (JinHai-CN) [inline core/src/server/web_impl/controller/WebController.hpp]: why not switch case?

**Review by XuPeng-SH — APPROVED** [review]

File: core/src/server/web_impl/handler/WebRequestHandler.h [hunk 2]
```diff
@@ -82,15 +82,18 @@ class WebRequestHandler {
         return context_ptr;
     }
 
+ protected:
+    Status
+    GetTaleInfo(const std::string& table_name, TableFieldsDto::ObjectWrapper& table_fields);
```
[the remark this answers is not in the document]
Comment (BossZou) [reply core/src/server/web_impl/handler/WebRequestHandler.h]: raname as `GetTableInfo`

File: core/src/server/web_impl/controller/WebController.hpp (hunk 1 above)
Comment (BossZou) [reply core/src/server/web_impl/controller/WebController.hpp]: Now the code is:
```
switch (status_dto->code->getValue()) {
            case StatusCode::SUCCESS:
                response = createDtoResponse(Status::CODE_200, partition_list_dto);
                break;
            case StatusCode::TABLE_NOT_EXISTS:
                response = createDtoResponse(Status::CODE_404, status_dto);
                break;
            default:response = createDtoResponse(Status::CODE_400, status_dto);
        }
```

File: core/src/server/web_impl/handler/WebRequestHandler.cpp [hunk 3]
```diff
@@ -81,81 +81,146 @@ WebErrorMap(ErrorCode code) {
...
+        size_t i = 0;
+        param->ids->forEach([&id_array, &i](const OInt64& item) { id_array[i++] = item->getValue(); });
     }
 
     return Status::OK();
 }
 
 Status
-CopyRowRecords(const SearchRequestDto::ObjectWrapper& param, engine::VectorsData& vectors) {
+CopyRowRecords(const SearchRequestDto::ObjectWrapper& param, engine::VectorsData& vectors, bool bin = false) {
```
[the remark this answers is not in the document]
Comment (BossZou) [reply core/src/server/web_impl/handler/WebRequestHandler.cpp]: The share code is placed in util.cpp

File: core/src/server/web_impl/handler/WebRequestHandler.cpp [hunk 4]
```diff
@@ -637,15 +750,28 @@ WebRequestHandler::DropPartition(const OString& table_name, const OString& tag)
...
+    if (!status.ok()) {
+        ASSIGN_RETURN_STATUS_DTO(status)
+    }
+
+    auto metric = engine::MetricType(schema.metric_type_);
     engine::VectorsData vectors;
-    auto status = CopyRowRecords(param, vectors);
-    if (status.code() == SERVER_INVALID_ROWRECORD_ARRAY) {
-        RETURN_STATUS_DTO(BODY_FIELD_LOSS, "Field \'records\' is required to fill vectors")
+    if (engine::MetricType::HAMMING == metric || engine::MetricType::JACCARD == metric ||
```
[the remark this answers is not in the document]
Comment (BossZou) [reply core/src/server/web_impl/handler/WebRequestHandler.cpp]: Save it.
Now is :
```
bool bin_flag = engine::MetricType::HAMMING == metric || engine::MetricType::JACCARD == metric ||
                    engine::MetricType::TANIMOTO == metric;

    if (!bin_flag) {
        if (nullptr == request->records.get()) {
            RETURN_STATUS_DTO(BODY_FIELD_LOSS, "Field \'records\' is required to fill vectors");
        }
        vectors.vector_count_ = request->records->count();
        status = CopyRowRecords(request->records, vectors.float_data_);
    } else {
        if (nullptr == request->records_bin.get()) {
            RETURN_STATUS_DTO(BODY_FIELD_LOSS, "Field \'records_bin\' is required to fill vectors");
        }
        vectors.vector_count_ = request->records_bin->count();
        status = CopyBinRowRecords(request->records_bin, vectors.binary_data_);
    }
```

**Review by del-zhenwu — APPROVED** [review]

**Review by cydrain — APPROVED** [review]

**Pull request merged**